core/src/main/java/com/linecorp/armeria/common/DefaultFlagsProvider.java

```diff
@@ -97,6 +97,7 @@ final class DefaultFlagsProvider implements FlagsProvider {
     static final String DNS_CACHE_SPEC = "maximumSize=4096";
     static final long DEFAULT_UNLOGGED_EXCEPTIONS_REPORT_INTERVAL_MILLIS = 10000;
     static final long DEFAULT_HTTP1_CONNECTION_CLOSE_DELAY_MILLIS = 3000;
+    static final long DEFAULT_CLIENT_FACTORY_GRACEFUL_SHUTDOWN_TIMEOUT_MILLIS = 10000;
```

Comment:
I wasn't sure if this value was sensible. 😅 A shorter value should be needed for rapid restart and a longer value is useful for a more graceful shutdown.

Reply:
I think we can probably do it when we reach 2.0:
- Move `ServerBuilder.DEFAULT_GRACEFUL_SHUTDOWN_TIMEOUT` to `Flags`
- Increase `ServerBuilder.DEFAULT_GRACEFUL_SHUTDOWN_TIMEOUT` value from zero
- Use the value for the client factory as well.

Reply:
Sounds good. Let me create an issue based on your idea.

Reply:
I agree with @jrhee17. We should not wait this long. I'd actually prefer this graceful shutdown feature to be opt-in only.

Reply:
The value of 10 seconds was derived from `Server`'s the default request timeout. Since many users want to gracefully shut down the server, they will set the graceful shutdown timeout to a value longer than the request timeout. They don't want to fail requests that come in just before the health check goes down. So, I set the default client factory to 10 seconds, the same as the default request timeout.

I think this part is controversial, so I would like to discuss it further.

Currently the server's graceful shutdown is disabled for this operation, so opt-in seems to be more sensible. However, I would recommend that most Line service users enable `defaultClientFactoryGracefulShutdownTimeoutMillis()` option, so I question whether opt-in is truly a good thing.

Reply:
For prod, yeah we need it, but for testing and dev, where devs spent most of their time, we probably don't need it? So I'd say we should keep it opt-in only unless we have a notion of 'profile' or 'environment'.

Reply:
That is correct. I wasn't sure about the default behavior either. We need to think more about this issue and decide on a direction.

>  unless we have a notion of 'profile' or 'environment'.

It sounds like a good idea. We may provide a different `FlagsProvider` depending on `profile`.